tests/components/conftest.py

```diff
@@ -533,11 +533,13 @@ async def _async_handle_step(
         else:
             return result
 
-        if (
-            result["type"] is FlowResultType.ABORT
-            and flow.source != SOURCE_SYSTEM
-            and flow.source not in DISCOVERY_SOURCES
-        ):
+        # Exclude discovery flows - but only for first step
+        if flow.source == SOURCE_SYSTEM or flow.source in DISCOVERY_SOURCES:
+            if not hasattr(flow, "__user_visible"):
+                setattr(flow, "__user_visible", True)
```

Comment:
As suggested on Discord, consider changing the name of the attribute to better match the purpose (detect if our fixture has seen this flow before) and improving the comment to explain why we exclude discovery flows the first time.